rfcs/003-randomness.md

```diff
@@ -0,0 +1,193 @@
+colluding with 2/3s of the validators [DOT1][7] (at which point, you have
+bigger concerns than a single SP not proving their files).
+
+### BABE Randomness
+
+As part of BABE consensus, VRFs are used to select the validator that will
+propose a block — the VRF [randomness is exposed in the relay chain proof][1],
+meaning we just need to "pull" that randomness into our parachain; this is
+the approach taken by Moonbeam ([MB1][2], [MB2][3]), and StorageHub ([SH1][4],
+[SH2][5]).
+
+This approach involves injecting the BABE randomness value into parachain
+storage, then, when necessary fetch it for usage. The main advantage is that
+this approach is verifiable, the main disadvantage is that the randomness
+takes a long time to refresh as it is dependent on BABE epochs (which last
+4 hours on mainnet), meaning that we are unable to generate unpredictable
+values for 4 hours.
+
+BABE offers a AuthorVRFRandomness which, in short, consists of having the
+block author run a VRF over some inputs from the block being built [BABE1][8].
+
+### Local-VRF
+
+The Local-VRF approach is based on the ideas behind BABE's AuthorVRFRandomness,
+however, it seems to use a default hash value as input for the randomness
+[VRF1][9]. This approach is simpler to implement and understand, having the
+block author build on the previous VRF also reduces the attack surface for
+manipulation effectively creating a chain much like Filecoin's.
+
+However, depending on implementation, this approach may be biasable. If the
+inputs are manipulatable by the block author, they may generate ones that benefit
+them, however, similar to the block hash approach, a large part of the network
+must collude to effectively break the guarantees of this approach.
+
+<!-- NOTE: now re-reviewing this -->
+This approach also carries the issue that "non-use-case-specific" VRF
+implementations are not common, while VRFs are in production use across
+blockchains, however none of them are "plug and play".
+
+### drand Pallet
+
+The drand network is the randomness source of multiple blockchain networks,
+however, Polkadot is not one of them. Building a bridge to drand is not a
+trivial task, but some folks have started building just that. The
+[drand pallet][6] provides randomness from drand's Quicknet while being
+pallet-randomness compliant.
+
+Validation of the pulses can be done as part of the proof of validity of the
+parachain block, as such, if some collator submits a "fake" pulse, the validator
+should be able to detect and punish the collator accordingly.
+
+The main issue with the crate is that it is not ready for production use.
+
+## Conclusion
+
+This document covers multiple randomness methods, however, none of them
+are without issues. Thus, moving forward with a definite method seems unwise,
+the proposal I put forward is for the team to discuss and vote on a way
+forward, keeping in mind that this choice may and should most likely be changed
+as the randomness landscape evolves.
```

Comment:
I think this paragraph is redundant, in my opinion this is implicit. No matter what you say, it's an RFC and we're in software engineering. We need to make decisions and sometimes they change.